src/Features/Cipher/Autokey/AutokeyStream.tsx

```diff
@@ -1,19 +1,43 @@
 import { AutoKeyString } from 'puzzle-lib';
 import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { RootState } from '../../../Store/rootReducer';
 import KeyedCipherStream from '../KeyedCipherStream';
+import { clear, setConversion, setSecret, setText } from './autokeyCipherSlice';
 
-interface Props {
+const mapStateToProps = (state: RootState) => ({
+  conversion: state.cipher.autokey.conversion,
+  secret: state.cipher.autokey.secret,
+  text: state.cipher.autokey.text,
+});
+const mapDispatchToProps = {
+  clear,
+  setConversion,
+  setSecret,
+  setText,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+interface Props extends ConnectedProps<typeof connector> {
   prompt: JSX.Element | string;
 }
 
 function AutokeyStream(props: Props) {
   return (
     <KeyedCipherStream
       cipher={new AutoKeyString()}
+      conversion={props.conversion}
       id="AutokeyStream"
+      secret={props.secret}
```

Comment:
Fun fact: I originally called this property `key` without realizing that `key` is a special property in React. Took quite a while to debug that issue...